### PR TITLE
[IMP] base_vat: replace vatnumver by stdnum library.

### PR DESCRIPTION
**Pull request by jva-odoo**
Python module vatnumber doesn't seem maintained anymore. Therefore, we should:
    - call directly stdnum (which is maintained and mostly used everywhere in vatnumber)

Also improve stdnum import, vat fix method and vat expected formats

task-1915371

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -36,7 +36,7 @@ qrcode==5.3
 reportlab==3.3.0
 requests==2.20.0
 zeep==3.1.0
-vatnumber==1.2
+python-stdnum==1.11
```
Comment (jco-odoo) [inline requirements.txt]: Need to check if this is the correct version

File: addons/base_vat/models/res_partner.py [hunk 2]
```diff
@@ -67,42 +17,51 @@
...
-    'mx': 'ABC123456T1B',
+    'mx': 'MXGODE561231GR8 or GODE561231GR8',
     'nl': 'NL123456782B90',
     'no': 'NO123456785',
     'pe': 'PER10254824220 or PED10254824220',
     'pl': 'PL1234567883',
     'pt': 'PT123456789',
     'ro': 'RO1234567897',
+    'rs': 'RS101134702',
+    'ru': 'RU123456789047',
```
Comment (jco-odoo) [inline addons/base_vat/models/res_partner.py]: @hpr-odoo From where did you get these changes?

File: requirements.txt (hunk 1 above)
Comment (Yenthe666) [reply requirements.txt]: Why not 1.12? See https://pypi.org/project/python-stdnum/1.12/

**Comment by jco-odoo** [comment]
@jva-odoo Can you also check (in the meantime) if you have any answers to these questions?

File: requirements.txt (hunk 1 above)
Comment (jco-odoo) [reply requirements.txt]: @Yenthe666 I think it needs to be the one of the last debian/ubuntu stable by default

Comment (Yenthe666) [reply requirements.txt]: Ahh yes that makes sense. Would need a check, it's probably lower than the latest :)

File: addons/base_vat/models/res_partner.py (hunk 2 above)
Comment (jva-odoo) [reply addons/base_vat/models/res_partner.py]: @jco-odoo from stdnum library

File: requirements.txt (hunk 1 above)
Comment (jco-odoo) [reply requirements.txt]: @julienlegros In ubuntu 18.04 LTS, the version is 1.8, so would we need to put 1.8 here then?  I would actually prefer to use a newer version as Yenthe suggests as it supports certain things that are important to us.  (but then we probably need to change below too)

Comment (jco-odoo) [reply requirements.txt]: and it is sure, it is not python3-stdnum?

Comment (julienlegros) [reply requirements.txt]: Unless we need a specific feature belonging to a more recent version, the preferred approach is to require what's in the latest Debian, i.e. `python-stdnum==1.10`. Ubuntu 20.04 will ship with at least 1.12.

Comment (jco-odoo) [reply requirements.txt]: @hpr-odoo @jva-odoo Can you still adapt this to 1.10 as Julien said?

Comment (hpr-odoo) [reply requirements.txt]: @jco-odoo , changed to 1.10. Thank you

**Comment by pedrobaeza** [comment]
Coming to this through codetriage.com. Is this still relevant?

**Comment by jco-odoo** [comment]
@pedrobaeza You bet it is.  @qdp-odoo   Let's remove this crap from Odoo.

**Comment by qdp-odoo** [comment]
@jco-odoo can you rebase, sir?

**Comment by qdp-odoo** [comment]
@odony this touches the framework, doesn't it? you might want to be aware of it

File: requirements.txt [hunk 3]
```diff
@@ -40,7 +40,7 @@ qrcode==6.1
 reportlab==3.5.13
 requests==2.21.0
 zeep==3.2.0
-vatnumber==1.2
+python-stdnum==1.10
```
Comment (odony) [inline requirements.txt]: Any chance stdnum 1.8 is enough as a base dep? 1.10 is available in Debian 10 LTS, but Ubuntu will only have it in April 2020 with 20.04 LTS. We can always upgrade the dep to 1.10 after April 2020.

From what I can see in the [changelog](https://github.com/arthurdejong/python-stdnum/blob/master/NEWS), the main VAT-related change we have would be the Venezuela VAT number (RIF) which we could perhaps implement ourselves until then?

File: setup.py [hunk 4]
```diff
@@ -51,7 +51,7 @@
         'reportlab',  # windows binary pypi.python.org/pypi/reportlab
         'requests',
         'zeep',
-        'vatnumber',
+        'python-stdnum',
```
Comment (odony) [inline setup.py]: The list in kinda sorted, so perhaps you can put it in the right spot (even if some are misordered)
Also need to check with @d-fence what needs to be updated in other setup files, if there is anything.

Comment (d-fence) [reply setup.py]: python3-stdnum should be added in debian/control file in depends. Pay attention that they are also ordered.
If vatnumber is not used anymore, it should be removed from debian/control, setup.cfg, setup/package.dfdebian, setup/package.dffedora.
python3-stdnum already exists in setup.cfg, so no other changes are needed.

Comment (jco-odoo) [reply setup.py]: @jbw-odoo

Comment (jbw-odoo) [reply setup.py]: @d-fence It's done. I also removed vatnumber/added stdnum in :
odoo/addons/point_of_sale/tools/posbox/overwrite_before_init/etc/init_posbox_image.sh
odoo/setup/win32/winpy_requirements.txt

Comment (d-fence) [reply setup.py]: Thanks, LGTM from the packaging POV.

File: requirements.txt (hunk 3 above)
Comment (jbw-odoo) [reply requirements.txt]: @odony @qdp-odoo 
-> Changed to 1.8
I checked all countries vat number checks in vatnumber 1.2, stdnum 1.8 & odoo (this PR)
To preserve the previous behavior I :
- left the methods for countries already existing in Odoo (maybe check later if pertinent to remove them and use stdnum methods instead when they exists). Those override stdnum.
- copied methods for countries supported in vatnumber but not in stdnum 1.8 (for co, ua & ru) in odoo

Venezuela was not supported before, so I left it that way.

**Comment by qdp-odoo** [comment]
@robodoo r+

**Pull request closed without merging**